test/extended/operators/operators_status.go

```diff
@@ -0,0 +1,196 @@
+package operators
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	"github.com/stretchr/objx"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	syncStatusWait = 2 * time.Minute
+)
+
+var _ = g.Describe("[Feature:Platform][Disruptive] Managed cluster should", func() {
+	defer g.GinkgoRecover()
+
+	g.It("reconcile cluster operator status", func() {
+		cfg, err := e2e.LoadConfig()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		c, err := e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		dc, err := dynamic.NewForConfig(cfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// presence of the CVO namespace gates this test
+		g.By("checking for the cluster version operator")
+		skipUnlessCVO(c.Core().Namespaces())
+
+		// gate on all clusteroperators being ready
+		available := make(map[string]struct{})
+		g.By(fmt.Sprintf("waiting for all cluster operators to be stable at the same time"))
```

Comment:
Should there be a helper function for this rather than inlining it twice here and once in #22058?